src/vm/runners/cairo_runner.zig

```diff
@@ -26,6 +26,76 @@ const expectEqual = std.testing.expectEqual;
 const expectError = std.testing.expectError;
 const expectEqualSlices = std.testing.expectEqualSlices;
 
+/// Tracks the step resources of a cairo execution run.
+const RunResources = struct {
+    // We consider the 'default' mode of RunResources having infinite steps.
+    n_steps: ?usize = null,
+
+    pub fn consumed(self: *RunResources) bool {
+        if (self.n_steps) |n_steps| {
+            return n_steps == 0;
+        }
+
+        return false;
+    }
+
+    pub fn consumeStep(self: *RunResources) void {
+        if (self.n_steps) |n_steps| {
+            if (n_steps > 0) {
+                self.n_steps = n_steps - 1;
+            }
+        }
+    }
+};
+
+/// This interface is used in conditions where vm execution needs to be constrained by a certain amount of steps.
+/// It is primarily used in the context of Starknet and implemented by HintProcessors.
+const ResourceTracker = struct {
+    // define interface fields: ptr,vtab
+    ptr: *anyopaque, //ptr to instance
+    vtab: *const VTab, //ptr to vtab
+    const VTab = struct {
+        consumed: *const fn (ptr: *anyopaque) bool,
+        consumeStep: *const fn (ptr: *anyopaque) void,
+    };
+
+    /// Returns true if there are no resource-steps available.
+    pub fn consumed(self: ResourceTracker) bool {
```

Comment:
Maybe we could define a `Self` type at the beginning of the struct to avoid repetitions of structure name, wdt?